docs/taskcluster-yml-v1.md

```diff
@@ -74,20 +76,42 @@ tasks:
       ...
 ```
 
+NOTE: A well-designed template should produce `tasks: []` for any unrecognized `task_for` values; this allows later expansion of this service to handle more events.
+
 ### Branch Filtering
 
 You can also add a branch clause to your `$if` - `then` statement so that the task will only run for events on certain branches. For example, the task defined below will only run for pushes to the master branch:
 
 ```yaml
 version: 1
 tasks:
-  - $if: ' event.head.branch == "master" '
+  - $if: 'tasks_for == "github-push"'
     then:
-      ...
-      ...
+      $if: 'event.ref == "refs/heads/master"'
+      then:
+        ...
+        ...
 ```
 
-You can have both events and branches in your `$if` - `then` statement.
+Note that it is wise to always check `tasks_for` first in a conditional like this.
+Other event types do not have `event.ref`, which would lead to a template error if not for the `tasks_for` check.
+
+NOTE: Once JSON-e supports [short-circuit boolean operators](https://github.com/taskcluster/json-e/issues/244), these conditionals can be collapsed into one.
+
+### Tags
+
+Tag pushes can be identified as follows:
+
+```yaml
+version: 1
+tasks:
+  - $if: 'tasks_for == "github-push"'
+    then:
+      $if: 'event.ref[:10] == "refs/tags/"'
```

Comment:
what does `[:10]` mean here?

Reply:
It's a slice, and it takes the prefix of the ref to compare.  The whole expression is equivalent to `event.ref.startswith('refs/tags/')`, but there's no `startswith` operator in JSON-e.

Reply:
wow, that's cool!